src/tests/ftest/security/cont_overwrite_acl.py

```diff
@@ -0,0 +1,180 @@
+        super(OverwriteContainerACLTest, self).setUp()
+        self.daos_cmd = self.get_daos_command()
+        self.prepare_pool()
+        self.add_container(self.pool)
+
+    def test_acl_overwrite_invalid_inputs(self):
+        """
+        JIRA ID: DAOS-3708
+
+        Test Description: Test that container overwrite command performs as
+            expected with invalid inputs in command line and within ACL file
+            provided.
+
+        :avocado: tags=all,pr,security,container_acl,cont_overwrite_acl
+        """
+        # Get list of invalid ACL principal values
+        invalid_acl_filename = self.params.get("invalid_acl_filename", "/run/*")
+
+        # Check for failure on invalid inputs.
+        for acl_file in invalid_acl_filename:
+            try:
+                self.daos_cmd.container_overwrite_acl(
+                    self.pool.uuid,
+                    self.pool.svc_ranks[0],
+                    self.container.uuid,
+                    acl_file)
+            except process.CmdError as err:
+                if "No such file or directory" in err.result.stdout:
+                    self.log.info(
+                        "Found expected error %s with invalid acl-file %s",
+                        err.result.stdout,
+                        acl_file)
+                else:
+                    self.fail("overwrite-acl seems to have failed with \
+                        unexpected error: {}".format(err))
+
+        self.fail("container overwrite-acl command expected to fail.")
+
+    def test_overwrite_invalid_acl_file(self):
+        """
+        JIRA ID: DAOS-3708
+
+        Test Description: Test that container overwrite command performs as
+            expected with invalid inputs in command line and within ACL file
+            provided.
+
+        :avocado: tags=all,pr,security,container_acl,cont_overwrite_acl
+        """
+        acl_filename = "test_acl_file.txt"
+        invalid_file_content = self.params.get(
+            "invalid_acl_file_content", "/run/*")
+
+        for content in invalid_file_content:
+            path_to_file = os.path.join(os.getcwd(), acl_filename)
+            create_acl_file(path_to_file, content)
+
+            # Run overwrite command
+            try:
+                self.daos_cmd.container_overwrite_acl(
+                    self.pool.uuid,
+                    self.pool.svc_ranks[0],
+                    self.container.uuid,
+                    path_to_file)
+            except process.CmdError as err:
+                if "-1003" in err.result.stdout:
+                    self.log.info(
+                        "Found expected error %s with invalid file content %s",
+                        err.result.stdout, content)
+                else:
+                    self.fail("overwrite-acl seems to have failed with \
+                        unexpected error: {}".format(err))
+
+    @fail_on(CommandFailure)
+    def test_overwrite_valid_acl_file(self):
```

Comment:
I think in both good and bad cases, it would be valuable to check the results. I.e. daos cont get-acl to check whether the ACL was unchanged (in the expected failure case) or was changed to the expected value on success.

Reply:
Will do.